ios/ReactNativeBrownfield.swift

```diff
@@ -111,29 +121,54 @@ class ReactNativeBrownfieldDelegate: RCTDefaultReactNativeFactoryDelegate {
    * @param launchOptions Launch options, typically passed from AppDelegate.
    */
   @objc public func startReactNative(onBundleLoaded: (() -> Void)?, launchOptions: [AnyHashable: Any]?) {
-    guard reactNativeFactory == nil else { return }
-    
-    delegate.dependencyProvider = RCTAppDependencyProvider()
-    self.reactNativeFactory = RCTReactNativeFactory(delegate: delegate)
+    guard !hasStartedReactNative else { return }
+    _ = launchOptions
+    _ = factory
     
     if let onBundleLoaded {
       self.onBundleLoaded = onBundleLoaded
-      if RCTIsNewArchEnabled() {
-        NotificationCenter.default.addObserver(
-          self,
-          selector: #selector(jsLoaded),
-          name: NSNotification.Name("RCTInstanceDidLoadBundle"),
-          object: nil
-        )
-      } else {
-        NotificationCenter.default.addObserver(
-          self,
-          selector: #selector(jsLoaded),
-          name: NSNotification.Name("RCTJavaScriptDidLoadNotification"),
-          object: nil
-        )
+      let notificationName: Notification.Name = RCTIsNewArchEnabled()
+        ? Notification.Name("RCTInstanceDidLoadBundle")
+        : Notification.Name("RCTJavaScriptDidLoadNotification")
+
+      NotificationCenter.default.addObserver(
+        self,
+        selector: #selector(jsLoaded),
+        name: notificationName,
+        object: nil
+      )
+
+      if let bridge = reactNativeFactory?.bridge, !bridge.isLoading {
+        DispatchQueue.main.async { [weak self] in
+          self?.jsLoaded(Notification(name: notificationName))
+        }
       }
```

Comment:
This is changing the previous behavior. jsLoaded should be called by React Native sending this notification